src/autosize.js

```diff
@@ -109,7 +109,9 @@ function assign(ta) {
 		const docTop = document.documentElement && document.documentElement.scrollTop; // Needed for Mobile IE (ticket #240)
 
 		ta.style.height = '';
+        changeOverflow("scroll");
 		ta.style.height = (ta.scrollHeight+heightOffset)+'px';
+        changeOverflow("hidden");
 
```

Comment:
Actually, you can capture the scrollHeight before resetting the height of the textarea.

The issue is that we reset the height so it may remove any scrollbar we have from the parent, which will cause the textarea to be a little bit wider (around 17px wider I think, value to be changed based on the browser). 

When we set the height, we calculate the scrollHeight a second time but we are basing it on the fact that there is no scrollbar so the textarea is a little bit wider.

By capturing the scrollHeight before resetting the height, we get the proper value as the parent scrollbar is included.


```suggestion
    // we capture the scroll height based on the parent (with the parent scrollbar if required)
    const scrollHeight = ta.scrollHeight;

    // we reset the textarea's height and potentially hide the parent scrollbar
		ta.style.height = '';
		
		// we set the new height
		// the initial code was fetching the scrollHeight value again but was incorrect because of the previous reset
		ta.style.height = (scrollHeight+heightOffset)+'px';
```

The reason why I think this is a more robust solution is that we take the parent's scrollbar into account and not the textarea's scrollbar. 

Both solutions will work in most cases but we can now style scrollbars. However, your solution may fail if the parent scrollbar is different than the textarea scrollbar, the computation would be incorrect again.

cf: https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-width and https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-scrollbar